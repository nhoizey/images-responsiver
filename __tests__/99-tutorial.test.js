'use strict';

const prettier = require('prettier');
const imagesResponsiver = require('../index');

function cleanHtml(html) {
	return prettier.format(html, { parser: 'html' });
}

describe('tutorial 2', () => {
	test('selector', () => {
		const content = `<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>images-responsiver</title>
    <link rel="stylesheet" href="assets/styles.css" />
  </head>

  <body>
    <div class="container">
      <h1>images-responsiver</h1>
      <p><img src="assets/my-logo.png" alt="My logo" class="logo"></p>
      <p><code>images-responsiver</code> is <strong>a simple solution for most responsive images needs</strong>. It transforms simple <code>&lt;img src=&quot;â€¦&quot; /&gt;</code> HTML tags into better responsive images syntax with <code>srcset</code> and <code>sizes</code> attributes.</p>
      <p>Here is a simple image:</p>
      <p><img src="assets/colorful-netherlands.jpg" alt="A photo of colorful houses in Groningen, The Netherlands"></p>
      <p>Download the photo for free <a href="https://unsplash.com/photos/0jj1xNsSOBo">on Unsplash</a>.</p>
    </div>
  </body>
</html>
`;
		const transformed = imagesResponsiver(content);

		expect(cleanHtml(transformed)).toMatchSnapshot();
	});
});
