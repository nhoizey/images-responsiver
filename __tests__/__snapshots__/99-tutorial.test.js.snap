// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tutorial 2 selector 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <title>images-responsiver</title>
    <link rel=\\"stylesheet\\" href=\\"assets/styles.css\\" />
  </head>

  <body>
    <div class=\\"container\\">
      <h1>images-responsiver</h1>
      <p>
        <img
          data-pristine=\\"assets/my-logo.png\\"
          sizes=\\"100vw\\"
          srcset=\\"
            assets/my-logo-320.png   320w,
            assets/my-logo-560.png   560w,
            assets/my-logo-800.png   800w,
            assets/my-logo-1040.png 1040w,
            assets/my-logo-1280.png 1280w
          \\"
          src=\\"assets/my-logo-640.png\\"
          alt=\\"My logo\\"
          class=\\"logo\\"
        />
      </p>
      <p>
        <code>images-responsiver</code> is
        <strong>a simple solution for most responsive images needs</strong>. It
        transforms simple <code>&lt;img src=\\"â€¦\\" /&gt;</code> HTML tags into
        better responsive images syntax with <code>srcset</code> and
        <code>sizes</code> attributes.
      </p>
      <p>Here is a simple image:</p>
      <p>
        <img
          data-pristine=\\"assets/colorful-netherlands.jpg\\"
          sizes=\\"100vw\\"
          srcset=\\"
            assets/colorful-netherlands-320.jpg   320w,
            assets/colorful-netherlands-560.jpg   560w,
            assets/colorful-netherlands-800.jpg   800w,
            assets/colorful-netherlands-1040.jpg 1040w,
            assets/colorful-netherlands-1280.jpg 1280w
          \\"
          src=\\"assets/colorful-netherlands-640.jpg\\"
          alt=\\"A photo of colorful houses in Groningen, The Netherlands\\"
        />
      </p>
      <p>
        Download the photo for free
        <a href=\\"https://unsplash.com/photos/0jj1xNsSOBo\\">on Unsplash</a>.
      </p>
    </div>
  </body>
</html>
"
`;
